server/fleet/hosts.go

```diff
@@ -708,14 +708,6 @@ func (h *Host) IsEligibleForDEPMigration(isConnectedToFleetMDM bool) bool {
 func (h *Host) NeedsDEPEnrollment(isConnectedToFleetMDM bool) bool {
 	return h.MDMInfo != nil &&
 		!h.MDMInfo.Enrolled &&
-		// as a special case for migration with user interaction, we
-		// also check the information stored in host_mdm, and assume
-		// the host needs migration if it's not Fleet
-		//
-		// this is because we can't always rely on nano setting
-		// `nano_enrollment.active = 1` since sometimes Fleet won't get
-		// the checkout message from the host.
-		(!isConnectedToFleetMDM || h.MDMInfo.Name != WellKnownMDMFleet) &&
 		h.IsDEPAssignedToFleet()
 }
```

Comment:
in https://github.com/fleetdm/fleet/pull/19860 this was the diff, but we don't want any of those checks, we just need to know if it's unenrolled.

<img width="1093" alt="image" src="https://github.com/fleetdm/fleet/assets/4419992/6cb38832-01ca-41fe-abc1-2220b089b898">